### PR TITLE
update doc: NNI core capacities in Overview.md, standalone mode in Trials.md

### PR DESCRIPTION
**Pull request by QuanluZhang**
describe core capacities of NNI, including hyperparameter tuning, neural architecture search, model compression, and feature engineering.

describe standalone mode, in which trial code can run without NNI experiment for easy debug.

### CODE REVIEW

File: docs/en_US/Overview.md [hunk 1]
```diff
@@ -46,6 +46,38 @@ For each experiment, user only needs to define a search space and update a few l
 
 More details about how to run an experiment, please refer to [Get Started](Tutorial/QuickStart.md).
 
+## Core Capacities
```
Comment (squirrelsc) [inline docs/en_US/Overview.md]: Core features?

File: docs/en_US/Overview.md [hunk 2]
```diff
@@ -46,6 +46,38 @@ For each experiment, user only needs to define a search space and update a few l
...
+
+### Automatic Feature Engineering
+Automatic feature engineering is for users to find the best features for the following tasks. Detailed description and usage can be found [here](FeatureEngineering/Overview.md). It is supported through NNI trial SDK, which means you do not have to create an NNI experiment, but to import a built-in auto-feature-engineering algorithm in your trial code and simply run your trial code. 
+
+The auto-feature-engineering algorithms usually have a bunch of hyperparameters themselves. If you want to automatically tune those hyperparameters, you can choose a tuning algorithm (i.e., tuner) and start an NNI experiment for it.
+
+### General NAS Framework
+This NAS framework is for users to easily specify candidate neural architectures, for example, could specify multiple candidate operations (e.g., separable conv, dilated conv) for a single layer, and specify possible skip connections. NNI will find the best candidate automatically. On the other hand, the NAS framework provides simple interface for another type of users (e.g., NAS algorithm researchers) to implement new NAS algorithms. Detailed discription and usage can be found [here](NAS/Overview.md)
```
Comment (squirrelsc) [inline docs/en_US/Overview.md]: typo: discription 
There should be a period at end of last sentence.

File: docs/en_US/Overview.md [hunk 3]
```diff
@@ -46,6 +46,38 @@ For each experiment, user only needs to define a search space and update a few l
 
 More details about how to run an experiment, please refer to [Get Started](Tutorial/QuickStart.md).
 
+## Core Capacities
+
+Hyperparameter tuning is a core capacity of NNI, it can be used to tune hyperparameters in machine learning and deep learning models, parameters in complex systems, etc. While hyperparameter tuning is only one step in designing a machine learning or deep learning model. Thus, NNI is evolved to help more on finding a better model with the following (pipelined) components.
+
+<p align="center">
+<img src="https://user-images.githubusercontent.com/16907603/70137129-3f549700-16c8-11ea-8204-3202b90f24a9.png" alt="drawing" width="700"/>
```
Comment (squirrelsc) [inline docs/en_US/Overview.md]: The graph is confusing to me,
1. It looks the whole process can apply to all kinds of work, but actually not. For example, feature selections mostly applies to traditional machie learning algorithm, but NAS, model compression applies to deep learning.
2. what does the different colors mean? And is it important to highlight with different colors?

File: docs/en_US/Overview.md [hunk 4]
```diff
@@ -46,6 +46,38 @@ For each experiment, user only needs to define a search space and update a few l
 
 More details about how to run an experiment, please refer to [Get Started](Tutorial/QuickStart.md).
 
+## Core Capacities
+
+Hyperparameter tuning is a core capacity of NNI, it can be used to tune hyperparameters in machine learning and deep learning models, parameters in complex systems, etc. While hyperparameter tuning is only one step in designing a machine learning or deep learning model. Thus, NNI is evolved to help more on finding a better model with the following (pipelined) components.
```
Comment (squirrelsc) [inline docs/en_US/Overview.md]: IMO, NNI provides the capability to run multiple instances in parallel to find best combinations of parameters, especially the parameters are used during initialization. It's key capability. This feature can be used in various domains, like find best hyperparameters for a deep learning model, or find best configuration for database and other complex system with real data.
NNI is also like to provide algorithm toolkits for machine learning especially...

File: docs/en_US/Overview.md [hunk 5]
```diff
@@ -46,6 +46,38 @@ For each experiment, user only needs to define a search space and update a few l
...
+
```
Comment (xuehui1991) [inline docs/en_US/Overview.md]: why the `parameter tuning` is blue and others are green?

File: docs/en_US/TrialExample/Trials.md [hunk 6]
```diff
@@ -127,6 +127,14 @@ In the YAML configure file, you need to set *useAnnotation* to true to enable NN
 useAnnotation: true
 ```
 
+## Standalone mode for debug
+
+NNI supports standalone mode for trial code to run without starting an NNI experiment. This is for finding out bugs in trial code more conveniently. NNI annotation natively supports standalone mode, as the added NNI related lines are comments. For NNI trial APIs, in standalone mode, some APIs return fake/empty values, for example, `nni.get_next_parameter` will return `{}` (NOTE: for users please assign default values to the hyperparameters in your trial code). And some other APIs, such as `nni.report_final_result`, do not really report values.
```
Comment (xuehui1991) [inline docs/en_US/TrialExample/Trials.md]: some API? I think we should make it clear.

File: docs/en_US/Overview.md (hunk 1 above)
Comment (QuanluZhang) [reply docs/en_US/Overview.md]: sure.

File: docs/en_US/Overview.md (hunk 3 above)
Comment (QuanluZhang) [reply docs/en_US/Overview.md]: good point, will update this figure and description

File: docs/en_US/Overview.md (hunk 5 above)
Comment (QuanluZhang) [reply docs/en_US/Overview.md]: thanks, will update this figure

File: docs/en_US/TrialExample/Trials.md (hunk 6 above)
Comment (QuanluZhang) [reply docs/en_US/TrialExample/Trials.md]: thanks, will make it clear

File: docs/en_US/Overview.md (hunk 4 above)
Comment (QuanluZhang) [reply docs/en_US/Overview.md]: updated, thanks

File: docs/en_US/Overview.md (hunk 5 above)
Comment (QuanluZhang) [reply docs/en_US/Overview.md]: removed the figure

File: docs/en_US/Overview.md [hunk 7]
```diff
@@ -46,6 +46,33 @@ For each experiment, user only needs to define a search space and update a few l
 
 More details about how to run an experiment, please refer to [Get Started](Tutorial/QuickStart.md).
 
+## Core Features
```
Comment (scarlett2018) [inline docs/en_US/Overview.md]: I'm thinking to replace this or put this ahead of the current "easy-to-use", "scalability", etc. section. let's discuss more in our scrum today.

File: docs/en_US/Tutorial/HowToDebug.md [hunk 8]
```diff
@@ -86,3 +86,4 @@ A common example of this would be run the mnist example without installing tenso
 
 As it shows, every trial has a log path, where you can find trial'log and stderr.
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/en_US/Tutorial/HowToDebug.md]: fixed, thanks

File: docs/en_US/Tutorial/HowToDebug.md [hunk 9]
```diff
@@ -86,3 +86,4 @@ A common example of this would be run the mnist example without installing tenso
 
 As it shows, every trial has a log path, where you can find trial'log and stderr.
 
+**You can use [standalone mode](../TrialExample/Trials.md#standalone-mode-for-debug) to debug your trial code.**
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/en_US/Tutorial/HowToDebug.md]: replaced, thanks

**Pull request merged**